be/src/exec/parquet_writer.cpp

```diff
@@ -443,9 +480,333 @@ Status ParquetWriterWrapper::_write_one_row(TupleRow* row) {
                 break;
             }
             default: {
-                std::stringstream ss;
-                ss << "unsupported file format: " << _output_expr_ctxs[index]->root()->type().type;
-                return Status::InvalidArgument(ss.str());
+                fmt::memory_buffer err_ss;
+                fmt::format_to(err_ss, "unsupported file format: : {}.",
+                               (*_output_vexpr_ctxs)[index]->root()->type().type);
+                return Status::InvalidArgument(fmt::to_string(err_ss.data()));
+            }
+            }
+        }
+    } catch (const std::exception& e) {
+        LOG(WARNING) << "Parquet write error: " << e.what();
+        return Status::InternalError(e.what());
+    }
+    return Status::OK();
+}
+
+Status ParquetWriterWrapper::_write_one_row(vectorized::Block& block, size_t row) {
+    int num_columns = _output_vexpr_ctxs->size();
```

Comment:
It's better to write a util method to convert vectorized::Block to arrow::RecordBatch, and then write to file stream block by block.

Reply:
Now this implementation is based on row, the performance may not be good, so it need to be change write data by column after have another plan about vectorized.